tables/basic-table/step9.md

```diff
@@ -0,0 +1,59 @@
+In this step, we'll find and fix a bug that you may or may not have already noticed in the current implementation, and then properly mark our callbaks that use state properties as react "callback hooks" so that they work as expected within the react data change detection ecosystem that is hooks.
+
+## Task
+
+1) Set the per per dropdown to "2 per page" and navigate to the very last page using the pagination controls, now select "3 per page" from the dropdown and notice our rows are gone!
+
+2) Locate the `handlePerPageSelect` handler function and replace the call to `updateList(currentPage);` with the following logic.
+
+<pre class="file" data-target="clipboard">
+    // when a user reaches end of pagination, and sets numPerPage to a value
+    // that requires more data than we have available, ensure the rows are set according
+    // to the previous page's index (currentPage - 1)
+    if (value > numPerPage && currentPage === defaultRows.length / numPerPage) {
+      updateList(currentPage - 1);
+    } else {
+      updateList(currentPage);
+    }
+</pre>
+
+3) Locate the dependency array for the useEffect hook you created earlier (`[numPerPage]`). Register the handlePerPageSelect function as a dependency of the function, right alongside of numPerPage. So changing `[numPerPage]` to `[numPerPage, handlePerPageSelect]`. This tells React to keep these values in sync.
+
```

Comment:
I shuffled this around a bit, and included a link to some docs where I thought it would be helpful. I think by going ahead and registering both dependencies initially and making an additional note about it, it's less confusing. Then in later steps when we convert handlers to callback hooks we're only dealing with that single task. Let me know if you think this is better

Reply:
I like this better.